### PR TITLE
Shopping Cart - First Submission

### PR DESCRIPTION
**Pull request by oharace**
Ready for review.

### CODE REVIEW

File: shopping_cart/features/manage_shipping.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Feature: Manage Shipping
+  In order to view my shipping rates
+  As a customer
+  I want to input my address and view shipping estimates
+
+  Background:
+    Given I am on the shopping cart page
+
+  @happy
```
Comment (magicalbanana) [inline shopping_cart/features/manage_shipping.feature]: No need to tag the scenarios. Try to make scenarios that are expressive of the sad, bad and happy paths. The thing is, with integration/feature testing or UA testing the line between the three paths blurr. 

So it's best to write scenarios that are more expressive of what your intent is when it comes to your test.

File: shopping_cart/features/manage_cart.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Feature: Manage Cart
+  In order to not lose any items
+  As a customer
+  I want to save my items if I log in or out
+
+  Background:
+    Given I am on the shopping cart page
+
+  @happy
+  Scenario:
```
Comment (magicalbanana) [inline shopping_cart/features/manage_cart.feature]: Always have a title or summary for your scenario. This willl help the business people to understand what you are trying to do.

**Comment by magicalbanana** [comment]
You're on the right track. Though I would ask you to read the comments and resubmit.

File: shopping_cart/features/manage_shipping.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    When I input my address "346 S Boo Road, Apt. 101"
+    And I input my city "Chesterton"
+    And I input my state "IN"
+    And I input my zip code "46304"
+    Then I should see a shipping estimate for my address
+
+  @sad
+  Scenario: Input Invalid Shipping Address
+    When I input my address "This is incorrect road"
+    And I input my city "Fake"
```
Comment (paulghaddad) [inline shopping_cart/features/manage_shipping.feature]: Lines 20-22 are too low level. You can create one step encapsulating entering an invalid address:

`When I enter an invalid shipping address`

File: shopping_cart/features/manage_coupons.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @happy
+  Scenario: I add a valid unexpired coupon
+    When I add a coupon number to the coupon field
+    And the coupon is not expired
+    Then I should see an updated order cost with the discount included
+
+  @sad
+  Scenario: I add a valid expired coupon
+    When I add a coupon number to the coupon field
```
Comment (paulghaddad) [inline shopping_cart/features/manage_coupons.feature]: You can combine lines 17-18 into one step:

`When I add a valid coupon that is expired`

**Comment by paulghaddad** [comment]
@oharace Take a look at the comments and please resubmit. This is a good first iteration, but I'd like you to look at your Features and see if there are additional sad and bad paths. Also, take a look at your step definitions and see if some at too low level. I've pointed out a couple. Strive to write your step definitions at a higher level of abstraction.

See this article for reference: https://robots.thoughtbot.com/acceptance-tests-at-a-single-level-of-abstraction

**Pull request closed without merging**